internal/storage/distlock_test.go

```diff
@@ -0,0 +1,95 @@
+package storage
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func storageProviders(t *testing.T) map[string]StorageProvider {
+	providers := map[string]StorageProvider{
+		"MEMORY": &MEMStorage{},
+	}
+
+	if (os.Getenv("ETCD_HOST") != "") && (os.Getenv("ETCD_PORT") != "") {
+		providers["ETCD"] = &ETCDStorage{}
+	}
+
+	if (os.Getenv("POSTGRES_HOST") != "") && (os.Getenv("POSTGRES_PORT") != "") {
+		providers["POSTGRES"] = &PostgresStorage{}
+	}
+
+	return providers
+}
+
+func distLockProviders(t *testing.T) map[string]DistributedLockProvider {
+	providers := map[string]DistributedLockProvider{
+		"MEMORY": &MEMLockProvider{},
+	}
+
+	if (os.Getenv("ETCD_HOST") != "") && (os.Getenv("ETCD_PORT") != "") {
+		providers["ETCD"] = &ETCDLockProvider{}
+	}
+
+	if (os.Getenv("POSTGRES_HOST") != "") && (os.Getenv("POSTGRES_PORT") != "") {
+		providers["POSTGRES"] = &PostgresLockProvider{}
+	}
+
+	return providers
+}
+
+func TestInit(t *testing.T) {
+	for name, dl := range distLockProviders(t) {
+		err := dl.Init(nil)
+		if err != nil {
+			t.Errorf("DistLock Init() failed: %v for provider: %s", err, name)
+		}
+	}
+}
```

Comment:
IMO the best option for this is to do a one-off init in a `TestMain()` as mentioned above, which suffices to test this.

If we keep separate `Init()` calls in each test function, we can probably discard this and use https://pkg.go.dev/github.com/stretchr/testify@v1.10.0/require#NoError on the `Init()` returns elsewhere.